src/runtime/virtcontainers/clh_test.go

```diff
@@ -127,6 +127,25 @@ func TestCloudHypervisorAddVSock(t *testing.T) {
 	assert.Equal(clh.vmconfig.Vsock.Socket, "path")
 }
 
+func TestCloudHypervisorAddVfioDevice(t *testing.T) {
+	bdf := "02:10.1"
+
+	vfDevice := config.VFIOPCIDev{
+		BDF: bdf,
+	}
+	testCloudHypervisorAddDevice(t, vfDevice)
+}
+
+func testCloudHypervisorAddDevice(t *testing.T, structure interface{}) {
+	assert := assert.New(t)
+	clh := cloudHypervisor{}
+	switch s := structure.(type) {
+	case config.VFIODev:
+		err := clh.addVfioDevice(s)
+		assert.Nil(err)
```

Comment:
I think the the test will always pass as the new function `addVfioDevice()` will never return `error`. 

A more meaningful test should assert on whether the content of added VFIO device (e.g. `clh.vmconfig.DeviceConfig`) is as expected, say check if its `Path` and `Id` is execpted .